internal/services/costmanagement/view_resource_base.go

```diff
@@ -0,0 +1,586 @@
+package costmanagement
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/costmanagement/2022-10-01/views"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type costManagementViewBaseResource struct{}
+
+func (br costManagementViewBaseResource) arguments(fields map[string]*pluginsdk.Schema) map[string]*pluginsdk.Schema {
+	output := map[string]*pluginsdk.Schema{
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+
+		"chart_type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(views.PossibleValuesForChartType(), false),
+		},
+
+		"accumulated": {
+			Type:     pluginsdk.TypeBool,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"query": {
+			Type:     pluginsdk.TypeList,
+			MaxItems: 1,
+			Required: true,
+			Elem: &pluginsdk.Resource{
```

Comment:
Since we can have at most 1 query I'm wondering whether it could make sense to flatten this? Since there's a few levels of nesting in this block it would be good to reduce that if possible.

Reply:
I'm fine with that, the only thing I could think of is the possibility of having multiple graphs/queries in one View in the future. Will remove it for now :)